src/bp/ui-studio/src/web/views/OneFlow/sidePanel/TopicList/index.tsx

```diff
@@ -245,151 +271,173 @@ const TopicList: FC<Props> = props => {
+    } else {
+      const nodeData = el as NodeData
+      if (nodeData?.type === 'qna') {
+        props.editQnA(nodeData.name.replace('/qna', ''))
+      } else if (nodeData?.type !== 'addWorkflow' && type === 'document') {
+        props.setFocusedText(nodeData.name)
+      }
     }
   }
 
   const postProcessing = tree => {
     tree.forEach(parent => {
       parent.childNodes?.forEach(node => {
+        node.nodeData.topic = parent.id
         if (node.id === `${parent.id}/qna`) {
           const wfCount = parent.childNodes?.filter(parentNode => node.id !== parentNode.id).length
-          parent.label = (
-            <div className={style.topicName}>
-              {parent.label}{' '}
-              <span className={style.tag}>
-                {node.nodeData?.countByTopic} Q&A · {wfCount} WF
-              </span>
-            </div>
-          )
+          if (parent.id !== props.focusedText) {
+            parent.label = (
+              <div className={style.topicName}>
+                {parent.label}{' '}
+                <span className={style.tag}>
+                  {node.nodeData?.countByTopic} Q&A · {wfCount} WF
+                </span>
+              </div>
+            )
+          }
         }
       })
+
+      if (parent.id !== 'Built-In') {
+        parent.childNodes?.push({
+          id: 'addWorkflow',
+          name: '4/addWorkflow',
+          label: (
+            <div className={cx(style.treeNode, style.addWorkflowNode)}>
+              <span>{lang.tr('studio.flow.sidePanel.addWorkflow')}</span>
+            </div>
+          ),
+          parent,
+          icon: 'plus',
+          type: 'addWorkflow',
+          nodeData: {
+            type: 'addWorkflow',
+            topic: parent.id
+          }
+        })
+      }
     })
 
+    if (!props.filter) {
+      const separator = {
+        id: 'separator',
+        label: <hr />
+      }
+
+      tree.splice(tree.length - 1, 0, separator)
+    }
+
     return tree
   }
 
   const activeFlow = props.currentFlow?.name
   return (
-    <TreeView<NodeData>
-      elements={flows}
-      nodeRenderer={nodeRenderer}
-      folderRenderer={folderRenderer}
-      postProcessing={postProcessing}
-      onContextMenu={handleContextMenu}
-      onClick={onClick}
-      visibleElements={activeFlow && [{ field: 'name', value: activeFlow }]}
-      onDoubleClick={onDoubleClick}
-      filterText={props.filter}
-      pathProps="name"
-      filterProps="name"
-    />
+    <React.Fragment>
+      {flows.length <= 3 && (
+        <div className={style.topicsEmptyState}>
+          <div className={style.topicsEmptyStateBlock}>
+            <img width="70" src="assets/ui-studio/public/img/empty-state.svg" alt="Empty folder" />
+            <div className={style.topicsEmptyStateText}>{lang.tr('studio.flow.sidePanel.tapIconsToAdd')}</div>
+          </div>
+        </div>
+      )}
+      <div className={cx(style.tree, props.filter ? '' : style.unfilteredTree)}>
```

Comment:
Can you explain how that works? I'm affraid I'm not familiar with that flavor of javascript shortcut.

Reply:
it's part js and part classnames library. basically the `[style.unfilteredTree]` is the css class  but since it's a variable, you put it between `[]` and the `: props.filter` is your condition, if that condition isn't met, the css class won't be applied

Reply:
Implemented but you were missing a `!` before `props.filter`. Some of your others suggestions had bugs also.

Reply:
Quite possible, didn't try them, just saw possible improvements 😬